### ISSUE TITLE
Snapshot extension registration failed

### ISSUE THREAD

**Issue by lpf763827726** [bnb-chain/bsc#821]
#### System information

Geth version: `1.1.8-859186f2`
OS & Version: Linux
Commit hash : (if `develop`)

#### Actual behaviour
```
t=2022-03-30T05:00:57+0000 lvl=warn msg="Synchronisation failed, dropping peer"  peer=20719e52103b613a67d6a35f31c8e6359d8efd241fbcd21bf9dbc9fa344b9ba4 err=timeout
t=2022-03-30T05:00:57+0000 lvl=eror msg="Ethereum peer removal failed"           peer=20719e52 err="peer not registered"
t=2022-03-30T05:01:03+0000 lvl=info msg="Looking for peers"                      peercount=1 tried=34  static=6
t=2022-03-30T05:01:22+0000 lvl=eror msg="Snapshot extension registration failed" peer=fda76e15 err="peer connected on snap without compatible eth support"
t=2022-03-30T05:01:23+0000 lvl=eror msg="Snapshot extension registration failed" peer=fde637d5 err="peer connected on snap without compatible eth support"
t=2022-03-30T05:02:04+0000 lvl=info msg="Looking for peers"                      peercount=0 tried=30  static=6
t=2022-03-30T05:02:56+0000 lvl=info msg="Looking for peers"                      peercount=1 tried=40  static=6
t=2022-03-30T05:03:06+0000 lvl=info msg="Looking for peers"                      peercount=2 tried=82  static=6
t=2022-03-30T05:03:08+0000 lvl=info msg="Downloader queue stats"                 receiptTasks=0 blockTasks=372 itemSize="4.05 KiB" throttle=8192
t=2022-03-30T05:03:17+0000 lvl=info msg="Looking for peers"                      peercount=2 tried=74  static=6
t=2022-03-30T05:03:57+0000 lvl=eror msg="Snapshot extension registration failed" peer=dd2fa788 err="peer connected on snap without compatible eth support"
t=2022-03-30T05:03:57+0000 lvl=eror msg="Snapshot extension registration failed" peer=dd02a483 err="peer connected on snap without compatible eth support"
t=2022-03-30T05:04:19+0000 lvl=eror msg="Ethereum peer removal failed"           peer=20719e52 err="peer not registered"

```

#### Steps to reproduce the behaviour
```
./geth_linux --datadir ./bsc init ./genesis.json
./geth_linux --syncmode "full" --config ./config.toml --cache 18000 --port 30403 --allow-insecure-unlock --port "11544" --http --http.addr 0.0.0.0 --http.port 11545 --http.api db,eth,net,web3,personal,txpool --rpc.allow-unprotected-txs --txlookuplimit 0 --usb --datadir ./bsc
```

i can sync before, when i sync in block 16053149, it stuck

**Comment by yutianwu** [comment]
`peer connected on snap without compatible eth support` this error is normal, will not cause the sync issue. Is there any other errors?

**Comment by lpf763827726** [comment]
> `peer connected on snap without compatible eth support` this error is normal, will not cause the sync issue. Is there any other errors?

all error are here

**Comment by guagualvcha** [comment]
Would you try the latest config: https://github.com/bnb-chain/bsc/releases/download/v1.1.8/mainnet.zip?

**Comment by lpf763827726** [comment]
> Would you try the latest config: https://github.com/bnb-chain/bsc/releases/download/v1.1.8/mainnet.zip?

yes, i was use v1.1.8

**Comment by DigitalTokens** [comment]
INFO [04-04|04:12:44.883] IPC endpoint opened                      url=/bsc-data/geth.ipc
INFO [04-04|04:12:44.883] HTTP server started                      endpoint=127.0.0.1:8545 prefix= cors= vhosts=localhost
INFO [04-04|04:12:55.881] Looking for peers                        peercount=0 tried=18 static=0
INFO [04-04|04:12:56.036] New local node record                    seq=1,649,038,297,946 id=781c30e81b4cd0c5 ip=44.193.223.190 udp=30303 tcp=30303
INFO [04-04|04:13:06.214] Looking for peers                        peercount=0 tried=31 static=0
ERROR[04-04|04:13:08.060] Snapshot extension registration failed   peer=ecbeebb8 err="peer connected on snap without compatible eth support"
INFO [04-04|04:13:16.214] Looking for peers                        peercount=0 tried=38 static=0
INFO [04-04|04:13:26.322] Looking for peers                        peercount=0 tried=37 static=0
INFO [04-04|04:13:36.373] Looking for peers                        peercount=0 tried=33 static=0
ERROR[04-04|04:13:37.702] Snapshot extension registration failed   peer=63cbe74f err="peer connected on snap without compatible eth support"
ERROR[04-04|04:13:40.057] Snapshot extension registration failed   peer=1d4fc8e3 err="peer connected on snap without compatible eth support"

**Comment by DigitalTokens** [comment]
same errors even when using v1.1.8

**Comment by yutianwu** [comment]
> same errors even when using v1.1.8

this errors are normal, is there any other issues

**Comment by DigitalTokens** [comment]
what is the fix for this issue; when we need to sync/resume the node where it stopped.  goal here is not to start from scratch rather need to get delta data on geth resumee

**Comment by lpf763827726** [comment]
i try to keep syncing but still not work, and i try to syncing to a new folder, it's working

**Comment by DigitalTokens** [comment]
> i try to keep syncing but still not work, and i try to syncing to a new folder, it's working

syncing a new data folder means; your starting from scratch again.  same folder will allow to get delta data from where it stopped

**Comment by lpf763827726** [comment]
> > i try to keep syncing but still not work, and i try to syncing to a new folder, it's working
> 
> syncing a new data folder means; your starting from scratch again. same folder will allow to get delta data from where it stopped

yea, i know that, but i can't keep sync in my old folder, always stuck in block 16053149, it's impossible to wait here all the time, so i only can sync to a new folder again

**Comment by DigitalTokens** [comment]
> #### System information
> Geth version: `1.1.8-859186f2` OS & Version: Linux Commit hash : (if `develop`)
> 
> #### Actual behaviour
> ```
> t=2022-03-30T05:00:57+0000 lvl=warn msg="Synchronisation failed, dropping peer"  peer=20719e52103b613a67d6a35f31c8e6359d8efd241fbcd21bf9dbc9fa344b9ba4 err=timeout
> t=2022-03-30T05:00:57+0000 lvl=eror msg="Ethereum peer removal failed"           peer=20719e52 err="peer not registered"
> t=2022-03-30T05:01:03+0000 lvl=info msg="Looking for peers"                      peercount=1 tried=34  static=6
> t=2022-03-30T05:01:22+0000 lvl=eror msg="Snapshot extension registration failed" peer=fda76e15 err="peer connected on snap without compatible eth support"
> t=2022-03-30T05:01:23+0000 lvl=eror msg="Snapshot extension registration failed" peer=fde637d5 err="peer connected on snap without compatible eth support"
> t=2022-03-30T05:02:04+0000 lvl=info msg="Looking for peers"                      peercount=0 tried=30  static=6
> t=2022-03-30T05:02:56+0000 lvl=info msg="Looking for peers"                      peercount=1 tried=40  static=6
> t=2022-03-30T05:03:06+0000 lvl=info msg="Looking for peers"                      peercount=2 tried=82  static=6
> t=2022-03-30T05:03:08+0000 lvl=info msg="Downloader queue stats"                 receiptTasks=0 blockTasks=372 itemSize="4.05 KiB" throttle=8192
> t=2022-03-30T05:03:17+0000 lvl=info msg="Looking for peers"                      peercount=2 tried=74  static=6
> t=2022-03-30T05:03:57+0000 lvl=eror msg="Snapshot extension registration failed" peer=dd2fa788 err="peer connected on snap without compatible eth support"
> t=2022-03-30T05:03:57+0000 lvl=eror msg="Snapshot extension registration failed" peer=dd02a483 err="peer connected on snap without compatible eth support"
> t=2022-03-30T05:04:19+0000 lvl=eror msg="Ethereum peer removal failed"           peer=20719e52 err="peer not registered"
> ```
> 
> #### Steps to reproduce the behaviour
> ```
> ./geth_linux --datadir ./bsc init ./genesis.json
> ./geth_linux --syncmode "full" --config ./config.toml --cache 18000 --port 30403 --allow-insecure-unlock --port "11544" --http --http.addr 0.0.0.0 --http.port 11545 --http.api db,eth,net,web3,personal,txpool --rpc.allow-unprotected-txs --txlookuplimit 0 --usb --datadir ./bsc
> ```
> 
> i can sync before, when i sync in block 16053149, it stuck

any luck ? any fix yet ?

**Comment by Jolly23** [comment]
dropping peer is normal. But there is a bug about `Ethereum peer removal failed -- peer not registered`.
Fixed here #856
